CONTRIBUTING.md

```diff
@@ -71,14 +71,14 @@ If there are conflicts with jamulussoftware/jamulus showing when you raise your
 
 ```shell
 git remote add upstream git@github.com:jamulussoftware/jamulus ;# add the main repo as upstream remote if you didn’t before already to get the latest code
-git checkout master ;# checkout your local master
-git pull upstream master --tags ;# refresh from jamulussoftware/jamulus (i.e. the main repository)
-git push --tags --force ;# update your fork's master
-git checkout - ;# back to your working branch
-git rebase master ;# replay your changes onto the latest code (and fix conflicts if needed)
+git fetch upstream ; # Get the latest changes from the main upstream repo you added before (if needed)
+git checkout <theBranchYourWorkIsOn> ;# switch to (= checkout) the branch you want to rebase
+git rebase -i upstream/master ;# replay (=rebase) your changes onto the latest code (and fix conflicts if needed)
```

Comment:
Ok. Changed.

@hoffie @pgScorpio is this ok?

Reply:
> Not sure what the downsides of using git fetch

As far as I understand git fetch only gets the meta data, git pull gets all actual changes too.

> We should really try to document the simplest method here

I think we should document the **safest** method here, with minimal chance on conflicts/errors.

> git is powerful, but can also be tricky and we shouldn't write yet another generic git tutorial. :)

I totally agree, git can be very confusing and a lot of existing tutorials use different methods, and many of these methods are considered to be 'unsafe' in many discussions on the internet... So hard to tell which is better and very confusing for a newbie on git like me. So I think we should decide which is the best available tutorial and link to that one. (Though I still have no idea which one would be the best.)

Reply:
> As far as I understand git fetch only gets the meta data, git pull gets all actual changes too.

Most importantly, `git fetch` downloads the required objects. `git pull` does the same, but additionally modifiers your current branch by merging or rebasing, depending on the command line arguments. That doesn't sound like a downside per-se to me. :)

> I think we should document the safest method here, with minimal chance on conflicts/errors.

Yes.